drivers/sensor/adxl345/Kconfig

```diff
@@ -11,3 +11,146 @@ config ADXL345
 	select SPI if $(dt_compat_on_bus,$(DT_COMPAT_ADI_ADXL345),spi)
 	help
 	  Enable driver for ADXL345 Three-Axis Digital Accelerometer.
+
+if ADXL345
+
+choice
+	prompt "Trigger mode"
+	default ADXL345_TRIGGER_NONE
+	help
+	  Specify the type of triggering used by the driver.
+
+config ADXL345_TRIGGER_NONE
+	bool "No trigger"
+
+config ADXL345_TRIGGER_GLOBAL_THREAD
+	bool "Use global thread"
+	depends on GPIO
+	select ADXL345_TRIGGER
+
+config ADXL345_TRIGGER_OWN_THREAD
+	bool "Use own thread"
+	depends on GPIO
+	select ADXL345_TRIGGER
+
+endchoice
+
+config ADXL345_TRIGGER
+	bool
+
+config ADXL345_THREAD_PRIORITY
+	int "Thread priority"
+	depends on ADXL345_TRIGGER_OWN_THREAD && ADXL345_TRIGGER
+	default 10
+	help
+	  Priority of thread used by the driver to handle interrupts.
+
+config ADXL345_THREAD_STACK_SIZE
+	int "Thread stack size"
+	depends on ADXL345_TRIGGER_OWN_THREAD && ADXL345_TRIGGER
+	default 1024
+	help
+	  Stack size of thread used by the driver to handle interrupts.
+
+config ADXL345_ACTIVITY_THRESHOLD
```

Comment:
This and most of the following configs should be moved to devicetree properties to allow different values per device instance.

Reply:
Hi, I am also currently dealing with the ADXL345 and have found this PR, which would enable what I need from the accel. It seems like all ADXL accel sensors leverage the KConfig system to define the ACTIVITY values (ACTIVE, INACTIVE TIME, etc).